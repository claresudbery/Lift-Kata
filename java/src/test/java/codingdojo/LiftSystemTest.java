package codingdojo;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.approvaltests.Approvals.verify;

public class LiftSystemTest {

    @Test
    public void initialises_withTwoFloors_andSingleLift_asSpecified() {
        // Arrange
        Lift liftA = new Lift("A", 0);
        List<Lift> lifts = Collections.singletonList(liftA);
        List<Integer> floors = Arrays.asList(0, 1);
        List<Call> calls = Collections.emptyList();

        // Act
        LiftSystem liftSystem = new LiftSystem(floors, lifts, calls);
        liftSystem.tick();

        // Assert
        verify(new LiftSystemPrinter().print(liftSystem));
    }
}
