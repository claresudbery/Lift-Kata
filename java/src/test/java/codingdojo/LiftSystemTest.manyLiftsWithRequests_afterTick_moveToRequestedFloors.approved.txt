3                * 3
2           *      2
1      *        [C]1
0     [A]  [B]     0
...
3              [*C]3
2         [*B]     2
1    [*A]          1
0                  0
